### PR TITLE
Added test that verifies rcu_nocbs kernel argument on the node

### PR DESCRIPTION
**Pull request by dshchedr**
Also added some fixes for cpu affinity test

### CODE REVIEW

File: functests/1_performance/cpu_management.go [hunk 1]
```diff
@@ -59,7 +61,7 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 
 		Expect(profile.Spec.CPU.Reserved).NotTo(BeNil())
 		reservedCPU = string(*profile.Spec.CPU.Reserved)
-		reservedCPUSet, err := cpuset.Parse(reservedCPU)
+		reservedCPUSet, err = cpuset.Parse(reservedCPU)
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:64]: Good catch! This sees like is not related to the patch, I would move it to a different patch as a bugfix, same pr though.

File: functests/1_performance/cpu_management.go [hunk 2]
```diff
@@ -18,6 +19,7 @@ import (
 	. "github.com/onsi/ginkgo"
 	"github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
+
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:22]: Is the empty line created by an automated code formatter?

File: functests/1_performance/cpu_management.go [hunk 3]
```diff
@@ -72,7 +74,7 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			Expect(capacityCPU - allocatableCPU).To(Equal(int64(len(listReservedCPU))))
 		})
 
-		It("[test_id:28026][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Verify CPU affinity mask, CPU reservation and CPU isolation on worker node", func() {
+		It("[test_id:27081][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Verify CPU affinity mask, CPU reservation and CPU isolation on worker node", func() {
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:77]: Question so I can understand: why is the test id changed?

File: functests/1_performance/cpu_management.go [hunk 4]
```diff
@@ -91,16 +93,50 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			Expect(conf).To(MatchRegexp(fmt.Sprintf(`"reservedSystemCPUs": ?"%s"`, reservedCPU)))
 
 			By("checking CPU affinity mask for kernel scheduler")
-			cmd = []string{"/bin/bash", "-c", "taskset -pc $(pgrep 'rcu_sched|rcu_preempt')"}
+			cmd = []string{"/bin/bash", "-c", "taskset -pc 1"}
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:96]: I miss the "why" in the commit message, I am sure the change is needed, but a reason will help more :)

File: functests/1_performance/cpu_management.go [hunk 5]
```diff
@@ -91,16 +93,50 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
...
 			sched, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
 			Expect(err).ToNot(HaveOccurred(), "failed to execute taskset")
 			mask := strings.SplitAfter(sched, " ")
 			maskSet, err := cpuset.Parse(mask[len(mask)-1])
 			Expect(err).ToNot(HaveOccurred())
 
-			Expect(reservedCPUSet.IsSubsetOf(maskSet)).To(Equal(true))
+			Expect(reservedCPUSet.IsSubsetOf(maskSet)).To(Equal(true), "The init process (pid 1) should have right cpu affinity mask")
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:103]: What do you mean "right affinity mask" ?

File: functests/1_performance/cpu_management.go [hunk 6]
```diff
@@ -91,16 +93,50 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
...
 		})
 
+		It("[test_id:34358] Verify rcu_nocbs kernel argument on the node", func() {
+			By("checking that cmdline contains rcu_nocbs with right value")
```
Comment (marcel-apf) [inline functests/1_performance/cpu_management.go:107]: what "right" value?

**Review by marcel-apf — COMMENTED** [review]
Looks good to me, I have mostly understanding comments

File: functests/1_performance/cpu_management.go (hunk 1 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:64]: :) I use this variable in new test, so not sure it's good idea to make several PRs for it

File: functests/1_performance/cpu_management.go (hunk 3 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:77]: Test_id is linked to test case in polarion. Here we had just wrong test_id. 
We may have another reasons for changing test_id, for example when we are updating test to new logic - we may have to create new test case, so test_id will be changed.

File: functests/1_performance/cpu_management.go (hunk 2 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:22]: hm, Goland did it itself..

File: functests/1_performance/cpu_management.go (hunk 4 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:96]: to have "more generic approach", will add it to commit message

File: functests/1_performance/cpu_management.go (hunk 5 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:103]: heh, yeah - let's update it with values :)

File: functests/1_performance/cpu_management.go (hunk 6 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:107]: same, will update it

File: functests/1_performance/cpu_management.go (hunk 1 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go:64]: I means several patches in the same PR, when looking at git logs will be easier to track bugs

File: functests/1_performance/cpu_management.go (hunk 2 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go:22]: ok

File: functests/1_performance/cpu_management.go (hunk 3 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go:77]: This is also better come as  a separate patch in the same PR. @MarSik do we have clear guidelines when to split patches or is OK as it is (I don't want @dshchedr to do more work if not necessary).

File: functests/1_performance/cpu_management.go (hunk 4 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go:96]: thanks

File: functests/1_performance/cpu_management.go (hunk 5 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go]: cool, thanks

File: functests/1_performance/cpu_management.go (hunk 6 above)
Comment (marcel-apf) [reply functests/1_performance/cpu_management.go:107]: +1

File: functests/1_performance/cpu_management.go (hunk 3 above)
Comment (MarSik) [reply functests/1_performance/cpu_management.go:77]: A short explanation in the commit message is enough I think. It fixes the same test if I see right.

File: functests/1_performance/cpu_management.go (hunk 1 above)
Comment (dshchedr) [reply functests/1_performance/cpu_management.go:64]: I created 2 commits: 
1st - is all about fixes for existing CPU affinity test (changed pid, removed colon and changed test_id)
2nd - adding new test

**Comment by yanirq** [comment]
@dshchedr  please rebase
